### PR TITLE
Bug fixes for v0.6.2

### PR DESCRIPTION
**Pull request by llucax**
This PR fixes a few issues in v0.6.1 and prepares for the release of v0.6.2.

- Update docs and template version
- Exclude benchmarks from the source distribution
- Replace `section-indexes` with `navigation.indexes`

### CODE REVIEW

**Review by Marenz — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -42,7 +42,7 @@ directory will be created with the generated project name. For example:
 cd ~/devel
 cookiecutter gh:frequenz-floss/frequenz-repo-config-python \
     --directory=cookiecutter \
-    --checkout v0.5.2
+    --checkout v0.6.2
```
Comment (christianparpart) [inline README.md:45]: Can we actually have a CI checker script that makes sure that this number is always pointing to the latest release?

**Review by christianparpart — COMMENTED** [review]
There were so many (!) textual string bug-fixes related to the version number, I really think it would make sense to create a CI checker script that ensures that the version number (e.g. `0.6.2` following the project name `frequenz-repo-config`) is always matching the latest release. That should help avoiding such issues in the future (of course, doesn't have to be part of this PR here, but I just see the benefit of checking that in the future).

File: README.md (hunk 1 above)
Comment (llucax) [reply README.md:45]: Even better, I just created this issue: https://github.com/frequenz-floss/frequenz-repo-config-python/issues/138

Comment (llucax) [reply README.md:45]: Although a check might be nice for the cookiecutter template, that's not solved by the docs macro. A PR with a CI check for that would be welcome!

Comment (llucax) [reply README.md:45]: I guess it should run on tags only, and check that the `pyproject.toml` template is using the version that is being built.

Comment (llucax) [reply README.md:45]: If not a PR, a issue would be also welcome.

**Pull request merged**